core/src/main/scala/chisel3/properties/Class.scala

```diff
@@ -165,16 +173,69 @@ object Class {
     // Bind the Property[ClassType] type for this Object.
     contextMod match {
       case rm: RawModule => {
-        rm.addCommand(DefObject(sourceInfo, obj))
+        rm.addCommand(DefObject(sourceInfo, obj, obj.className.name))
         classProp.bind(OpBinding(rm, Builder.currentWhen), SpecifiedDirection.Unspecified)
       }
       case cls: Class => {
-        cls.addCommand(DefObject(sourceInfo, obj))
+        cls.addCommand(DefObject(sourceInfo, obj, obj.className.name))
         classProp.bind(ClassBinding(cls), SpecifiedDirection.Unspecified)
       }
       case _ => throwException("Internal Error! Property connection can only occur within RawModule or Class.")
     }
 
     obj
   }
+
+  implicit class ClassDefinitionOps[T <: Class](definition: Definition[T]) {
+
+    /** Get a Property[ClassType] type from a Definition[Class].
+      *
+      * This is useful when a Property[ClassType] type is needed for references to instances of the Class.
+      *
+      * This method is safe, and should be used over unsafeGetReferenceType when possible.
+      */
+    def getPropertyType: Property[ClassType] = {
+      // Get the BaseModule for the Class this is a definition of.
+      val baseModule = definition.getInnerDataContext.getOrElse(
+        throwException("Internal Error! Class instance did not have an associated BaseModule.")
+      )
+
+      // Get a Property[ClassType] type from the Class name.
+      val classType = ClassType.unsafeGetClassTypeByName(baseModule.name)
+      Property[classType.Type]()
+    }
+  }
+
+  implicit class ClassInstanceOps[T <: Class](instance: Instance[T]) {
+
+    /** Get a reference to an Instance[Class] as a Property[ClassType] property type.
+      *
+      * This method allows Instances of Classes to be safely connected to Property[ClassType] ports, so the references
+      * can be passed through the hierarchy.
+      */
+    def getPropertyTypeReference: Property[ClassType] = {
```

Comment:
`getPropertyTypeReference` feels just a little too similar to `getPropertyType`. What about just `getPropertyReference` or `getPropertyValue`?